From cc0288cf99f6745e0155a920cb15dc4e19b451b9 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:50 -0400
Subject: [virt] kvm/mmu: remove unused parameter

Message-id: <1406040016-3289-76-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86154
O-Subject: [RHEL7 PATCH v2 075/221] KVM: MMU: remove unused parameter
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

vcpu in page_fault_can_be_fast() is not used so remove it

Signed-off-by: Xiao Guangrong <xiaoguangrong@linux.vnet.ibm.com>
Signed-off-by: Gleb Natapov <gleb@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

Conflicts:
 arch/x86/kvm/mmu.c [37f6a4e237 already in RHEL]
(cherry picked from commit e5552fd252763c74ce6a6c27c7873939062b5038)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index 12527fb..f86e179 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -2784,7 +2784,7 @@ exit:
  return ret;
 }
 
-static bool page_fault_can_be_fast(struct kvm_vcpu *vcpu, u32 error_code)
+static bool page_fault_can_be_fast(u32 error_code)
 {
  /*
   * Do not fix the mmio spte with invalid generation number which
@@ -2840,7 +2840,7 @@ static bool fast_page_fault(struct kvm_vcpu *vcpu, gva_t gva, int level,
  if (!VALID_PAGE(vcpu->arch.mmu.root_hpa))
   return false;
 
- if (!page_fault_can_be_fast(vcpu, error_code))
+ if (!page_fault_can_be_fast(error_code))
   return false;
 
  walk_shadow_page_lockless_begin(vcpu);
-- 
1.7.1